src/framework/db/acl.clj

```diff
@@ -0,0 +1,145 @@
+(ns framework.db.acl
+  (:require
+    [clojure.string :as str]))
+
+(defn collify
+  [x]
+  (if (coll? x) x (vector x)))
+
+(defn ->table-name
+  [table]
+  (cond->
+    (flatten (collify table))
+    coll? first
+    keyword? name))
+
+(defn insert-action
+  ([actions table-actions]
+   (reduce (fn [acc {:keys [table actions]}] (insert-action acc table actions))
+     actions
+     table-actions))
+  ([actions table action]
+   (let [t (->table-name table)
+         table-actions (first (filter #(#{t} (:table %)) actions))
+         old-actions (into #{} (:actions table-actions))]
+     (conj (remove #(= table-actions %) actions)
+       {:table t :actions (reduce conj old-actions (collify action))}))))
```

Comment:
I'm not sure, but the module only should expose the helpful stack of functions, that's some functions should be using **defn-** instead of **defn**. It's the user perspective to reduce ambiguity use them.

Perhaps these one should be exposing only:
`                            ->necessary-permits
                             sql-map->owns?
                              sql-map->wheres
                              insert-action
                              owns?
                              acl
                              reverse-table-aliases
                              reverse-column-aliases
`

IMO :)

Reply:
Fixed